packages/sync/src/links/compositeQueueLink.ts

```diff
@@ -1,11 +1,23 @@
 import { OfflineQueueLink, TYPE_MUTATION } from "./OfflineQueueLink";
 import { LocalDirectiveFilterLink } from "./LocalDirectiveFilterLink";
 import { DataSyncConfig } from "../config";
-import { ApolloLink, concat } from "apollo-link";
+import { ApolloLink, Operation } from "apollo-link";
+import { onError } from "apollo-link-error";
+import { RetryLink } from "apollo-link-retry";
 
 export const compositeQueueLink = (config: DataSyncConfig, filter?: TYPE_MUTATION): ApolloLink => {
   const offlineLink = new OfflineQueueLink(config, filter);
   offlineLink.openQueueOnNetworkStateUpdates();
+  const errorLink = onError(({ networkError, operation }) => {
+    if (networkError) {
+      operation.variables.__networkError = networkError;
```

Comment:
This is not that simple. There is like tons of corner cases here. For example 403 will most likely cause networkError, but it do not mean that we are offline and should enqueue stuff. 
https://github.com/apollographql/apollo-link/blob/master/packages/apollo-link-error/src/index.ts#L66